main.tf

```
@@ -9,6 +9,9 @@ locals {
   # If skip_final_snapshot is set to false then assemble a string with the prefix-rds_identifier-random_id. dec is used because b64 and id can contain characters that rds does not allow
   snapshot_identifier = var.skip_final_snapshot == false ? "${var.final_snapshot_prefix}${var.rds_identifier}-${random_id.rid.dec}" : ""
 
+  # Determine if var.create_subnet_group is true. If it is then use aws_db_subnet_group.rds[0].name. If it is false then use the provided var.subnet_group_name
+  db_subnet_group_name = var.create_subnet_group == true  ? aws_db_subnet_group.rds[0].name : (var.db_subnet_group_name != null ? var.db_subnet_group_name : null)   
```

```suggestion
  db_subnet_group_name = var.create_subnet_group  ? aws_db_subnet_group.rds[0].name : (var.db_subnet_group_name != null ? var.db_subnet_group_name : null)   
```
Since `?` is a true evaluator and we are checking the boolean value of `var.create_subnet_group` we shouldn't need the `var.create_subnet_group == true`

---

Good one, thanks, done.